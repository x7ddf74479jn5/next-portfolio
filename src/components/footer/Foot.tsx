import Link from "next/link";

type Props = {
  href: string;
  label: string;
};

const Foot: React.VFC<Props> = ({ href, label }) => {
  return (
    <Link href={href}>
      <a>{label}</a>
    </Link>
  );
};

export default Foot;
