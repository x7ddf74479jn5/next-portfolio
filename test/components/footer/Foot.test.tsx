import type { RenderResult } from "@testing-library/react";
import Foot from "src/components/footer/Foot";
import { render, setupWindow } from "test/test-utils";

describe("Foot", () => {
  let renderResult: RenderResult;
  setupWindow();
  const footItems = [
    {
      id: "item 1",
      href: "item 1",
      label: "item 1",
    },
    {
      id: "item 2",
      href: "item 2",
      label: "item 2",
    },
    {
      id: "item 3",
      href: "item 3",
      label: "item 3",
    },
  ];

  beforeEach(() => {
    renderResult = render(<Foot footItems={footItems} />);
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(renderResult.getByRole("list").children).toHaveLength(3);
    const listElement = renderResult.getAllByRole("listitem")[0];
    expect(listElement).toHaveTextContent("item 1");
    const anchorElement = renderResult.getAllByRole("link")[0];
    expect(anchorElement).toHaveAttribute("href", expect.stringMatching("item"));
  });
});
