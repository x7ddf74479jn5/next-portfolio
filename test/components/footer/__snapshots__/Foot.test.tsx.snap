// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foot matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="foot"
  >
    <ul>
      <li>
        <a
          href="/item%201"
        >
          item 1
        </a>
      </li>
      <li>
        <a
          href="/item%202"
        >
          item 2
        </a>
      </li>
      <li>
        <a
          href="/item%203"
        >
          item 3
        </a>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
